src/main/java/mods/battlegear2/asm/transformers/InventoryArrayAccessTransformer.java

```diff
@@ -0,0 +1,439 @@
+                            + classNode.name
+                            + "#"
+                            + targetMethod.clearMapping);
+        } else {
+            BattlegearLoadingPlugin.logger.error(
+                    "Expected " + expectedCount
+                            + " injections, but injected "
+                            + injectionCount
+                            + " times when transforming "
+                            + classNode.name
+                            + "#"
+                            + targetMethod.clearMapping);
+        }
+    }
+
+    /**
+     * <p>
+     * Wraps all the array access to the EntityPlayer's mainInventory with an if statement :
+     * </p>
+     *
+     * <pre>
+     * player.inventory.mainInventory[player.inventory.currentItem] = itemStack;
+     * </pre>
+     * <p>
+     * Replaced by :
+     * </p>
+     *
+     * <pre>
+     * if (!InventoryAccessHook.setPlayerCurrentItem(player, itemStack)) {
+     *     player.inventory.mainIventory[player.inventory.currentItem] = itemStack;
+     * }
+     * </pre>
+     */
+    private static int wrapInventoryAccess(InsnList instructions) {
+        int injectionCount = 0;
+        for (final AbstractInsnNode insnNode : instructions.toArray()) {
+            if (isLoadPlayerNode(insnNode)) {
+                final boolean isPlayerField = insnNode instanceof FieldInsnNode;
+                final AbstractInsnNode prevNode = insnNode.getPrevious();
+                if (isPlayerField && !(prevNode instanceof VarInsnNode && prevNode.getOpcode() == Opcodes.ALOAD
+                        && ((VarInsnNode) prevNode).var == 0)) {
+                    continue;
+                }
+                final AbstractInsnNode secondNode = insnNode.getNext();
+                if (isPlayerInventoryFieldNode(secondNode)) {
+                    final AbstractInsnNode thirdNode = secondNode.getNext();
+                    if (isMainInventoryFieldNode(thirdNode)) {
+                        final AbstractInsnNode fourthNode;
+                        if (isPlayerField) {
+                            if (!(thirdNode.getNext() instanceof VarInsnNode
+                                    && thirdNode.getNext().getOpcode() == Opcodes.ALOAD
+                                    && ((VarInsnNode) thirdNode.getNext()).var == 0)) {
+                                continue;
+                            }
+                            fourthNode = thirdNode.getNext().getNext();
+                        } else {
+                            fourthNode = thirdNode.getNext();
+                        }
+                        if (isLoadPlayerNode(fourthNode)) {
+                            final AbstractInsnNode fifthNode = fourthNode.getNext();
+                            if (isPlayerInventoryFieldNode(fifthNode)) {
+                                final AbstractInsnNode sixthNode = fifthNode.getNext();
+                                if (isCurrentItemFieldNode(sixthNode)) {
```

Comment:
This is very deeply nested, can you invert all the if conditions here and use `if(!isXYZ(node)) {continue;}`?

Reply:
I know that I can do thatn and I tried it, but I personally prefer keeping the nested statements statements, I found that more readable than inverting everything and using continue